torch/_meta_registrations.py

```diff
@@ -3645,6 +3645,39 @@ def meta_relu_(self):
 
 @register_meta([aten.index_put.default, aten._unsafe_index_put.default])
 def meta_index_put(self, indices, values, accumulate=False):
+    if not isinstance(indices, tuple):
+        indices = (indices,)
+
+    torch._check(
+        len(indices) > 0,
+        lambda: "Passing an empty index list to torch.index_put() is not valid syntax",
+    )
+    torch._check(
+        self.device == values.device,
+        lambda: "Tensor and value must be on the same device",
+    )
+
+    torch._check(
+        self.dim() >= len(indices),
+        lambda: f"too many indices for tensor of dimension {self.dim()} (got {len(indices)})",
+    )
+
+    torch._check(
+        len(indices) != 0 or utils.is_expandable_to(values.size(), self.size()),
```

Comment:
where is `len(indices) != 0` coming from? You are already checking that indices is non empty.